Roundstart Mime 🎭

AwkwardDryad opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->
[REWORK][FEATURE][QOL]

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
- adds mime as a roundstart role!
- added "mime holes" to most maps! (unintrusively located annexed onto existing clown holes or near them)
- added carnival cant! historically, carnival cant is a subset of local languages carnival workers would use to conceal secrets of the trade, but the cant in-game is a mime-inclusive sign language dialect! This allows the mime and clown to freely communicate with each other. everyone else will see gibberish and hand waving.

## Why's this needed? <!-- Describe why you think this should be added to the game. -->
- at the moment, mime is a surprisingly popular role that a lot of players have wanted to see brought to roundstart in a meaningful way.


## Changelog
<!-- If necessary, put your changelog entry below. Otherwise, please delete it.
Use however you want to be credited in the changelog in place of CodeDude.
Use (*) for major changes and (+) for minor changes. For example: -->

```changelog
(u)AwkwardDryad
(*)Mime is now a roundstart role!
(*)The mime and clown can now communicate with each other using their hands! How professional～
```

AwkwardDryad commented:
the destiny map render is a little strange... the clown's bed and helium tank was moved up a bit. the render didn't show that 😋

ZeWaka commented:
does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter

AwkwardDryad commented:
> does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter

hold my beer.

Sovexe commented:
> > does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter
> 
> hold my umm...drink.

the drink is invisible

AwkwardDryad commented:
> > > does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter
> > 
> > 
> > hold my umm...drink.
> 
> the drink is invisible

*tilts her head*

predmetparfuyit86 commented:
> does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter

Rancher doesnt have any unique traitor items either. though i guess chickens sort of makeup for that

AwkwardDryad commented:
i mean in all honesty i think a mime would do perfectly fine with the massive list of default traitor items (as well as some of the staffie items)

vode-code commented:
Give me time, and I will code mime traitor items.
Current ideas: a cursed mask I currently have in progress, a spatial mime outfit with the front pocket not sewn shut, and some sort of bloody scarf.

predmetparfuyit86 commented:
> i mean in all honesty i think a mime would do perfectly fine with the massive list of default traitor items (as well as some of the staffie items)

i agree but unique items is always cool like the clown has a TON of unique traitor items and i think it would be cool if mime also had a unique item or 2 maybe like an umbrella that fires a poison bullet (like the real life bulgarian umbrella)

BatElite commented:
While we wait for mime-specific items (I'm sure some can be thought of) I think we should at least give the mime access to boom boots in case they want to betray their silent ways in addition to the station.

AwkwardDryad commented:
> While we wait for mime-specific items (I'm sure some can be thought of) I think we should at least give the mime access to boom boots in case they want to betray their silent ways in addition to the station.

i just coded them a unique item and gave them access to the pickpocket gun and the butcher knife so far! of course more can be added at a later date. i don't want to bloat this patch too much with traitor items

vode-code commented:
"fingers strangley"
is this supposed to say they're strangling something?

AwkwardDryad commented:
> "fingers strangley"
> is this supposed to say they're strangling something?

oh god typo

AwkwardDryad commented:
> does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter

done :D

AwkwardDryad commented:
> Give me time, and I will code mime traitor items.
> Current ideas: a cursed mask I currently have in progress, a spatial mime outfit with the front pocket not sewn shut, and some sort of bloody scarf.

OMG I DIDNT SEE YOU WERE WORKING ON A CURSED MASK AAAA

predmetparfuyit86 commented:
> Give me time, and I will code mime traitor items.
> Current ideas: a cursed mask I currently have in progress, a spatial mime outfit with the front pocket not sewn shut, and some sort of bloody scarf.

the scarf should act like the fiber wire so you can strangle people with it

vode-code commented on code/mob/living/carbon/human.dm:
```diff
@@ -1430,7 +1430,11 @@
 
-	message = process_accents(src,message)
+	if (src.wear_mask && istype(src.wear_mask,/obj/item/clothing/mask/mime/cursed))
+		whisper(message, forced=TRUE)
```
this looks like it makes accents not apply?

AwkwardDryad replied:
accents are processed in whisper too. whisper is basically say 2......

AwkwardDryad replied:
oh shit i elsed it heh...

AwkwardDryad replied:
whisper code:
![Code_94YJ3cUBVH](https://user-images.githubusercontent.com/12144569/109736511-ea285f00-7b70-11eb-81ef-ce35ac324fcb.png)

AwkwardDryad replied:
i'm basically redirecting say to whisper and cutting it.

vode-code commented:
> 
> the scarf should act like the fiber wire so you can strangle people with it

The current idea I have is something like the demonic briefcase/pen, plus the fiber wire.
It would never be bloody, but would always be blood-stained, and whenever fresh blood from a new non-monkey blood DNA type is added, the defense would increase slightly. It would add an ability button that would act somewhat like the mounted defib or grabs, adding a "scarf grab" item to both your hands, and would cut someone whenever you grab them with it, adding to the blood pool.
I'm also thinking that the Suspenders should get a traitor item equivalent, perhaps a loud suspender snap?
Silencing scarf, deafening suspenders?

AwkwardDryad commented:
> > the scarf should act like the fiber wire so you can strangle people with it
> 
> The current idea I have is something like the demonic briefcase/pen, plus the fiber wire.
> It would never be bloody, but would always be blood-stained, and whenever fresh blood from a new non-monkey blood DNA type is added, the defense would increase slightly. It would add an ability button that would act somewhat like the mounted defib or grabs, adding a "scarf grab" item to both your hands, and would cut someone whenever you grab them with it, adding to the blood pool.
> I'm also thinking that the Suspenders should get a traitor item equivalent, perhaps a loud suspender snap?
> Silencing scarf, deafening suspenders?

heh...i love this!!! feel free to code any of these ideas btw. i'm just here to make mime a roundstart job 🥰

WALPVRGIS commented:
I love. All of this. TwT

Sovexe commented on code/datums/jobs.dm:
```diff
@@ -138,6 +138,47 @@
 					phrase_log.log_phrase("name-[ckey(src.name)]", M.real_name, no_duplicates=TRUE)
+				if((M.job == "Clown") || (M.job == "Mime"))
+					var/make_bracelet = FALSE //friendship bracelets!!!
+					for(var/mob/MOB in mobs)
```
this should either break after finding the first match or at least have a cap

if an admin sets the clown job limit high as a gimmick each person gets a bracelet for every other person. That's a lot of bracelets. May also apply to mass respawning people as clowns and mimes.

AwkwardDryad replied:
i was thinking about that :/

AwkwardDryad replied:
i set the limit to 4 sets of bracelets that can generate.

Sovexe commented on code/datums/abilities/carnival_cant.dm:
```diff
@@ -0,0 +1,36 @@
+        if(!message)
+            return
+        OWNER.emote("carnivalcant")
+        for(var/mob/LISTENER in viewers(OWNER,10))
```
believe you need to use `AIviewers` here since you want the ai to be able to hear this

AwkwardDryad replied:
ill test it, they should be able to hear though.

AwkwardDryad replied:
yep yep! works perfectly fine～

Sovexe commented on _std/defines/item.dm:
```diff
@@ -44,6 +44,8 @@
 #define TGUI_INTERACTIVE		 (1<<19)
 /// Has a click delay for attack_self()
 #define ATTACK_SELF_DELAY		 (1<<20)
+///can be put in the glove slot
+#define ONGLOVE (1<<21)
```
don't know that this really need to be a bit flag. Why not just make the bracelets a subtype of glove with `hide_prints = 0`

AwkwardDryad replied:
hmm, ill try it

AwkwardDryad replied:
done!

Sovexe commented on code/mob/living/carbon/human/procs/emote.dm:
```diff
@@ -2015,6 +2015,25 @@
+						src.pixel_x = old_x
+						src.pixel_y = old_y
+					var/chosen_gibberish = pick(gibberish)
+					maptext_out = chosen_gibberish
```
need to set a message as well if you want overhead text, otherwise this will only do the minor pixel shift and nothing else
```suggestion
					maptext_out = "<i>[chosen_gibberish]</i>"
					message = "<b>[src]</b> [chosen_gibberish]"
```

AwkwardDryad replied:
there is a message, but i *did* forget to italicise it :P

Sovexe replied:
When I tried it locally it didn't work at all until I added that message var

AwkwardDryad replied:
hmmm....eek you're right...

AwkwardDryad commented on code/datums/abilities/carnival_cant.dm:
```diff
@@ -0,0 +1,36 @@
+        var/mob/living/carbon/human/OWNER = holder.owner
+        if(!OWNER.limbs.l_arm && !OWNER.limbs.r_arm)
+            OWNER.show_text("You can't speak Carnival Cant without arms!","red")
+        var/message = input(OWNER, "What would you like to say?", "Carnival Cant") as null|text
```
the copytext should strip it, but ill check again

Sovexe replied:
Easy to test, just spam <hr> and view the line spam it creates

I believe there's literally just a strip_html proc you can use

AwkwardDryad replied:
done done!

AwkwardDryad commented:
five hours of sleep is suffering

Sovexe commented on code/datums/abilities/carnival_cant.dm:
```diff
@@ -0,0 +1,37 @@
+    regenRate = 0
+    tabName = "Carnival"
+
+/datum/targetable/carnival/cant //defaults to clown version
+    name = "Carnival Cant"
+    desc = "A secret language that only entertainers know! This one in particular is a form of sign language!"
+    icon_state = "cant-clown"
+    targeted = 0
+    cooldown = 0
+    pointCost = 0
```
This will keep the base type out of `concrete_typesof` and I suppose we might as well just slide up some vars while we're at it. Additionally sets our preferred holder type for carnival abilities

```suggestion
ABSTRACT_TYPE(/datum/targetable/carnival)
/datum/targetable/carnival
	cooldown = 0
	last_cast = 0
	pointCost = 0
	preferred_holder_type = /datum/abilityHolder/carnival

/datum/targetable/carnival/cant //defaults to clown version
    name = "Carnival Cant"
    desc = "A secret language that only entertainers know! This one in particular is a form of sign language!"
    icon_state = "cant-clown"
    targeted = 0

```

AwkwardDryad replied:
oooh thank you

Sovexe commented on code/datums/jobs.dm:
```diff
@@ -138,6 +138,51 @@
+					var/make_bracelet = FALSE //friendship bracelets!!!
+					var/friend_limit = 4
+					for(var/mob/MOB in mobs)
+						if((MOB != M) && ((MOB.job == "Clown") || (MOB.job == "Mime")))
```
Just thought of something else for this...should probably verify that the friend is alive (and maybe that they have a client)

also probably check for `hibernating = 1` - which is a var on `mob/living` set when they go into the cryotron

I can only imagine the sadness of a clown looking for a non-existent friend

that way you don't get bracelets for the dead or clowns in cryo

AwkwardDryad replied:
done～❤️

Sovexe reviewed (approved):
good artwork on this

LadyBirb commented:
> does the mime have traitor items? if not, i think having it as a roundstart job is a non-starter

better remove captain and hop from the game then

predmetparfuyit86 commented:
> > While we wait for mime-specific items (I'm sure some can be thought of) I think we should at least give the mime access to boom boots in case they want to betray their silent ways in addition to the station.
> 
> i just coded them a unique item and gave them access to the pickpocket gun and the butcher knife so far! of course more can be added at a later date. i don't want to bloat this patch too much with traitor items

in my eyes the mimes butcher knife shouldnt have the butcher ability because thats kind of the chefs thing

AwkwardDryad commented:
> in my eyes the mimes butcher knife shouldnt have the butcher ability because thats kind of the chefs thing

i  like this idea, so i made a generic butcher knife

Sovexe commented on code/obj/item/clown_items.dm:
```diff
@@ -85,3 +85,31 @@ VUVUZELA
+	hide_prints = FALSE
+	var/friend_name
+	var/mob/friend_mob
+
```
```suggestion

disposing()
	src.friend_mob = null
	. = ..()

```

AwkwardDryad replied:
ooh good call i forgot to do that 😋

flrsh commented:
Haven't looked at the code yet but I'd really prefer generally having just one clown and one mime. I like the fact that they're "the clown" and "the mime" and sort of become the defacto ideal of what a clown or mime is like.

AwkwardDryad commented:
> Haven't looked at the code yet but I'd really prefer generally having just one clown and one mime. I like the fact that they're "the clown" and "the mime" and sort of become the defacto ideal of what a clown or mime is like.

we had a pretty long discussion in space lobby a while ago and it went overwhelmingly in favor of two clowns and two mimes. its a pretty controversial change that myself and a lot of people wanted to see playtested live for a while. my idea was to bump it back down if people weren't enjoying it. i'm going to ask around again and see what the people online currently think, but in my personal opinion i think this is one of those things that's better livetested by the community

AwkwardDryad commented:
update: we ran a poll and it was...pretty split, but tipped over in favor of no. the general consensus after that is that jobs should scale off server population (which i 100% agree with), but thats another thing entirely, so i'll bump them back down to one

geneva-breaker commented on code/datums/syndicate_buylist.dm:
```diff
@@ -765,6 +774,14 @@ This is basically useless for anyone but miners.
 	blockedmode = list(/datum/game_mode/spy, /datum/game_mode/spy_theft, /datum/game_mode/revolution)
 
+/datum/syndicate_buylist/traitor/generic_butcherknife
+	name = "Beeg Nife (TM)"
```
Tone.

geneva-breaker replied:
no actually, this is weird to have in the buylist.

geneva-breaker commented on code/datums/jobs.dm:
```diff
@@ -138,6 +138,51 @@
 				M.choose_name(3, src.name, default)
 				if(M.real_name != default && M.real_name != orig_real)
 					phrase_log.log_phrase("name-[ckey(src.name)]", M.real_name, no_duplicates=TRUE)
+				if((M.job == "Clown") || (M.job == "Mime"))
```
So this doesn't look voluntary... I'm not sure how you're planning to address imposing "friendship" between several unrelated players (on RP or not) but this doesn't seem like a good idea.

ZeWaka commented on maps/clarion.dmm:
```diff
@@ -14514,6 +14514,9 @@
 /obj/decal/poster/wallsign/poster_clown{
 	pixel_y = 32
 	},
+/obj/landmark/start{
+	name = "Mime"
+	},
```
can't modify clarion without an exception

ZeWaka commented on code/obj/item/clothing/gimmick.dm:
```diff
@@ -984,6 +984,32 @@
+
+		afterattack(atom/target, mob/user, reach, params)
+			if(reach <= 1 && user.mind && user.mind.assigned_role == "Mime" && istraitor(user) && istype(user,/mob/living/carbon/human) && istype(target,/mob/living/carbon/human))
+				var/mob/living/carbon/human/ATTACKER = user
+				var/mob/living/carbon/human/DEFENDER = target
```
all caps local non-const variables 🤢

virvatuli-gh commented:
Opinions from a longtime mime player:

I do not like having my character (in this case my mime) being pushed by gameplay elements to act contrary to how I play that character. Dictating what the player should be doing, in this case saying "the clown and the mime are friends, they have a friendship bracelet and a secret communication method" is projecting what you want the mime player to be doing, not what they may or may not want to do.

If you want to suggest that the mime and clown are friends, you give hints at them being so, not drastic gameplay elements that push the player into assuming that identity. Put a framed picture of the clown and the mime on a table in the mime's room as an example, this lets a player create a story around that picture rather than just reading them the story out loud and assuming they want that story.

I do like the idea of the mime having more permanence on shifts, them being a real job with map permanence is cool. As with most jobs of those kinds, it's good to have interesting traitor items that both are a nod to the job they're exclusive to and are interesting both mechanically and visually. The two exclusive traitor items you've added, essentially the syndicate dagger and a cursed clown mask with slightly different flavour, I can't see ever being used really because they don't do anything that would make a mime player go "Heck yeah I'm a traitor mime, I should use these over a suppressed pistol or fibre wire".

This kind of PR is one that you should definitely have used the [FEEDBACK] tag for, as it drastically changes a job and I'm certain there are many players who don't frequent the discord nor github that would want to talk about potential improvements to their favourite job, and what they'd like to see or not see.

AwkwardDryad commented:
> This kind of PR is one that you should definitely have used the [FEEDBACK] tag for, as it drastically changes a job and I'm certain there are many players who don't frequent the discord nor github that would want to talk about potential improvements to their favourite job, and what they'd like to see or not see.

i've been talking with people on the discord a lot about this actually, and also, half of the patch wasn't my call in the first place. which is why i'm trimming it now.

AwkwardDryad commented:
> This kind of PR is one that you should definitely have used the [FEEDBACK] tag for, as it drastically changes a job and I'm certain there are many players who don't frequent the discord nor github that would want to talk about potential improvements to their favourite job, and what they'd like to see or not see.

Followup, i removed the friendship mechanic entirely.

AwkwardDryad commented:
> I do not like having my character (in this case my mime) being pushed by gameplay elements to act contrary to how I play that character. Dictating what the player should be doing, in this case saying "the clown and the mime are friends, they have a friendship bracelet and a secret communication method" is projecting what you want the mime player to be doing, not what they may or may not want to do.

> I do like the idea of the mime having more permanence on shifts, them being a real job with map permanence is cool. As with most jobs of those kinds, it's good to have interesting traitor items that both are a nod to the job they're exclusive to and are interesting both mechanically and visually. The two exclusive traitor items you've added, essentially the syndicate dagger and a cursed clown mask with slightly different flavour


I added the friendship bracelets because Emily said at one point that she wouldn't support roundstart mime unless there was more interaction with the clown. She suggested bracelets

I added super quick and flavorless mime traitor items because Zewaka would only support roundstart mime of the mime had traitor items.

None of these awful decisions were my call and I'm literally just trying to get mime in a position to be a roundstart job without having to fight 10 people for it like i did with fucking plastic silverware. Traitor items aren't the focus of the game and they can literally be added later at a n y point if people want traitor mimes to be more interesting, but that's not what this patch is about and they're an ad-lib to hopefully get this patch merged painlessly.
